File: sendBack/src/main/java/com/logistics/entity/Address.java
package com.logistics.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("t_address")
@Schema(description = "地址实体")
public class Address {

    @TableId(type = IdType.AUTO)
    @Schema(description = "地址ID")
    private Long id;

    @Schema(description = "联系人姓名")
    private String name;

    @Schema(description = "联系电话")
    private String phone;

    @Schema(description = "省份")
    private String province;

    @Schema(description = "城市")
    private String city;

    @Schema(description = "区县")
    private String district;

    @Schema(description = "详细地址")
    private String detail;

    @Schema(description = "地址标签(home-家,company-公司,school-学校)")
    private String tag;

    @Schema(description = "是否默认地址")
    private Integer isDefault;

    @Schema(description = "地址类型(sender-寄件,receiver-收件)")
    private String type;

    @Schema(description = "用户ID")
    private Long userId;

    @TableField(fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private LocalDateTime updatedAt;

    @TableLogic
    @Schema(description = "是否删除")
    private Integer deleted;
}

File: sendFront/src/api/address.ts
import { del, get, post, put } from './request'

// 地址类型
export type AddressType = 'sender' | 'receiver'

// 地址请求参数
export interface AddressDTO {
  name: string
  phone: string
  province: string
  city: string
  district: string
  detail: string
  tag?: string
  isDefault: boolean
  type: AddressType
}

// 地址响应（与后端一致）
export interface AddressVO {
  id: number
  name: string
  phone: string
  province: string
  city: string
  district: string
  detail: string
  fullAddress: string
  tag: string | null
  isDefault: boolean
  type: AddressType
  createdAt: string
}

// 获取地址列表
export function getAddressList() {
  return get<AddressVO[]>('/addresses')
}

// 根据类型获取地址列表
export function getAddressByType(type: AddressType) {
  return get<AddressVO[]>('/addresses', { params: { type } })
}

// 获取地址详情
export function getAddressById(id: number) {
  return get<AddressVO>(`/addresses/${id}`)
}

// 添加地址
export function addAddress(data: AddressDTO) {
  return post<AddressVO>('/addresses', data)
}

// 更新地址
export function updateAddress(id: number, data: Partial<AddressDTO>) {
  return put<AddressVO>(`/addresses/${id}`, data)
}

// 删除地址
export function deleteAddress(id: number) {
  return del<void>(`/addresses/${id}`)
}

// 设为默认地址
export function setDefaultAddress(id: number) {
  return put<AddressVO>(`/addresses/${id}/default`, {})
}
